src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,77 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
-    public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+    public <T> T getInstance(Class<T> interfaceClass) {
+        if (instances.containsKey(interfaceClass)) {
+            return interfaceClass.cast(instances.get(interfaceClass));
+        }
+
+        for (Class<?> clazz : getAllClasses()) {
+            if (interfaceClass.isAssignableFrom(clazz)
+                    && clazz.isAnnotationPresent(Component.class)) {
+                try {
+                    T instance = (T) createInstance(clazz);
+                    instances.put(interfaceClass, instance);
+                    return instance;
+                } catch (Exception e) {
+                    throw new RuntimeException(e);
```

Comment:
When catching exceptions, it is better to catch a more general exception like `ReflectiveOperationException` instead of just `Exception`. Moreover, include an informative message and the caught exception when throwing a new `RuntimeException`.